### ISSUE TITLE
Build failed with Debian stable / rustc 1.70

### ISSUE THREAD

**Issue by lscorcia** [Lonami/lzxd#30]
Hi,
when building this crate with `cargo build --release` on Debian Stable (rust 1.70), a bunch of the following errors occur:

```
error[E0446]: private type `DecodeFailed` in public interface
  --> /lzxd-0.2.5/src/bitstream.rs:42:5
   |
42 |     pub fn read_bit(&mut self) -> Result<u16, DecodeFailed> {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private type
   |
  ::: /lzxd-0.2.5/src/lib.rs:99:1
   |
99 | enum DecodeFailed {
   | ----------------- `DecodeFailed` declared as private
```

The fix seems easy enough - declare the DecodeFailed enum as public.

**Comment by Lonami** [comment]
Would you mind sending a PR?

**Comment by Lonami** [comment]
Couldn't even clone the damn repository due to #22:

```
>git clone git@github.com:Lonami/lzxd.git
Cloning into 'lzxd'...
remote: Enumerating objects: 448, done.
remote: Counting objects: 100% (70/70), done.
remote: Compressing objects: 100% (38/38), done.
remote: Total 448 (delta 47), reused 35 (delta 32), pack-reused 378 (from 1)
Receiving objects: 100% (448/448), 103.35 KiB | 661.00 KiB/s, done.
Resolving deltas: 100% (291/291), done.
Updating files: 100% (18/18), done.
Downloading testdata/8-renderdoc.lzx (21 MB)
Error downloading object: testdata/8-renderdoc.lzx (615e9d6): Smudge error: Error downloading testdata/8-renderdoc.lzx (615e9d66baf4a0f8266d4eeab24f8b6480ac3216269b9a74c85a4721eccb2233): batch response: This repository is over its data quota. Account responsible for LFS bandwidth should purchase more data packs to restore access.

Errors logged to 'C:\Users\L\Documents\Projects\lzxd\.git\lfs\logs\20241217T230924.6745543.log'.
Use `git lfs logs last` to view the log.
error: external filter 'git-lfs filter-process' failed
fatal: testdata/8-renderdoc.lzx: smudge filter lfs failed
warning: Clone succeeded, but checkout failed.
You can inspect what was checked out with 'git status'
and retry with 'git restore --source=HEAD :/'
```

So I've removed those files for now, and:

```
Published lzxd v0.2.6 at registry `crates-io`
```
